pandas/core/series.py

```diff
@@ -210,6 +210,11 @@ def __init__(self, data=None, index=None, dtype=None, name=None,
                     raise ValueError("cannot specify a dtype with a "
                                      "Categorical unless "
                                      "dtype='category'")
+                if index is not None and len(index) != len(data):
```

Comment:
this should go a little further down after
```
if index is None:
    ....
else:
      # this the check here
```

Reply:
though maybe this should go in ``_sanitize_array`` around L3242

Reply:
I don't see an advantage to moving it into `_sanitize_array` versus putting it in the if at L230.  But I could be missing something. What do you think?

Reply:
it’s prob ok here but a bit lower
early failure is good

Reply:
Ok. Proposed a placement in this next push. Placing it in the 
```
if index is None:
    ...
```
at ~L226 as an else breaks some tests if data is a scalar or `SingleBlockManager`. It's difficult to catch all cases if we put it in `_sanitize_array` because of the returns in the if cases.  So the next best place appears to be after the call to `_sanitize_array`. Not ideal with regard to early failure, but really the only place that I can see to put a single check just to be able to do `len(data)`.  Not much different from the current location except that it catches cases other than `Categorical`.